history.php

```diff
@@ -21,17 +21,36 @@
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
+use tool_clonecategory\form\clonecategoryhistory_form;
 use tool_clonecategory\history_table;
 
 require('../../../config.php');
 require_once($CFG->libdir . '/adminlib.php');
 
 admin_externalpage_setup('clonecategory_history');
 
+$start = optional_param('start', strtotime('-3 month', time()), PARAM_INT);
+$end = optional_param('end', time(), PARAM_INT);
+
+$clonehistoryform = new clonecategoryhistory_form(null, [
+   'startdate' => $start,
+   'enddate' => $end,
+]);
+
+// Redirect back if form is cancelled.
+if ($clonehistoryform->is_cancelled()) {
+    redirect(new moodle_url("/admin/tool/clonecategory/history.php"));
```

Comment:
Would it be a bit cleaner to pu `$PAGE->set_url` above this line, and then use `redirect($PAGE->url)` ?

Reply:
added in latest commit